Source/Core/VideoBackends/OGL/ProgramShaderCache.cpp

```diff
@@ -157,6 +162,9 @@ void ProgramShaderCache::UploadConstants()
 		memcpy(buffer.first + ROUND_UP(sizeof(PixelShaderConstants), s_ubo_align) + ROUND_UP(sizeof(VertexShaderConstants), s_ubo_align),
 			&GeometryShaderManager::constants, sizeof(GeometryShaderConstants));
 
+		memcpy(buffer.first + ROUND_UP(sizeof(PixelShaderConstants), s_ubo_align) + ROUND_UP(sizeof(VertexShaderConstants), s_ubo_align) + ROUND_UP(sizeof(GeometryShaderConstants), s_ubo_align),
```

Comment:
I don't think it's wise to include this buffer within the other uniforms. So it must be reuploaded on all uniform changes. It's likely better to create a new uniform buffer object for every UID, and to just bind it on every change.